overlord/snapstate/refreshhints_test.go

```diff
@@ -323,6 +323,60 @@ func (s *refreshHintsTestSuite) TestRefreshHintsStoresRefreshCandidates(c *C) {
 	c.Check(snapst, DeepEquals, &snapst2)
 }
 
+func (s *refreshHintsTestSuite) TestPruneRefreshCandidates(c *C) {
+	st := s.state
+	st.Lock()
+	defer st.Unlock()
+
+	c.Assert(snapstate.PruneRefreshCandidates(st, "some-snap"), IsNil)
+
+	candidates := map[string]*snapstate.RefreshCandidate{
+		"snap-a": {
+			SnapSetup: snapstate.SnapSetup{
+				Type: "app",
+				SideInfo: &snap.SideInfo{
+					RealName: "snap-a",
+					Revision: snap.R(1),
+				},
+			},
+		},
+		"snap-b": {
+			SnapSetup: snapstate.SnapSetup{
+				Type: "app",
+				SideInfo: &snap.SideInfo{
+					RealName: "snap-b",
+					Revision: snap.R(1),
+				},
+			},
+		},
+		"snap-c": {
+			SnapSetup: snapstate.SnapSetup{
+				Type: "app",
+				SideInfo: &snap.SideInfo{
+					RealName: "snap-c",
+					Revision: snap.R(1),
+				},
+			},
+		},
+	}
+	st.Set("refresh-candidates", candidates)
+
+	c.Assert(snapstate.PruneRefreshCandidates(st, "snap-a"), IsNil)
+
+	var candidates2 map[string]*snapstate.RefreshCandidate
+	c.Assert(st.Get("refresh-candidates", &candidates2), IsNil)
+	c.Check(candidates2["snap-a"], IsNil)
```

Comment:
Here I might show my golang ignorance, but isn't it better if we check that the second argument returned by the dictionary lookup is `false`?

I mean, something like (dunno if it can be written in a more compact way):
```golang
_, ok := candidates2["snap-a"]
c.Check(ok, Equals, false)
```
The difference is that (again, if I'm not misled) your code would also succeed if the `snap-a` is still left in the candidates map, just with a `nil` value.

Reply:
Yes, you are right in that it could potentially hide a problem where nils are left in the map, so I tweaked the check. Thanks.